### DISCUSSION TITLE
Can script command output to plain text instead of typescript?

### DISCUSSION THREAD

**Discussion by leontmn** [util-linux/util-linux#1734] [Q&A] [answered]
When running the script command to record terminal inputs and outputs. Is there a way to have the output in plain text format instead typescript?

The reason I asked is because the output file is not clean (includes control / weird characters) when opened in any text editor.

I used the command below to do most of the cleanup, but doesn't work well most of the time.
cat sample_rawlog.txt | perl -pe 's/\e([^\[\]]|\[.*?[a-zA-Z]|\].*?\a)//g' | col -b > clean.txt

[sample_rawlog.txt](https://github.com/util-linux/util-linux/files/9004327/sample_rawlog.txt)

**Comment by karelzak** [comment, marked as answer]
This use case is currently not supported.

We can try to implement it, but it will be pretty tricky to filter out all the "bad" chars as the script itself does not interpret the chars and has no clue what is control / weird char. The script is just a proxy between application and terminal, it reads and writes, but it has no clue what the content is :-)

**Comment by davidkitfriedman** [comment]
[`ansi2txt`](http://manpages.ubuntu.com/manpages/bionic/man1/ansi2txt.1.html) and [`ansi2html`](http://manpages.ubuntu.com/manpages/bionic/man1/ansi2html.1.html) can take a file created by [`script`](https://en.wikipedia.org/wiki/Script_(Unix)) and make it more easily readable.

A Debian package called [colorized-logs](https://packages.debian.org/search?keywords=colorized-logs&searchon=names&suite=all&section=all&sourceid=mozilla-search) provides these for Debian based distros. One could also do Google searches for [ansi2txt](https://www.google.com/search?&q=ansi2txt) or [ansi2html](https://www.google.com/search?&q=ansi2html) as there can be other implementations (e.g., [ansi2html as a Python Package](https://pypi.org/project/ansi2html/)).

(just learned about this in [course 4 of the Google IT Support Professional Certificate (paywall unfortunately)](https://www.coursera.org/learn/system-administration-it-infrastructure-services/supplement/8Tt3K/recording-your-actions))
